### PR TITLE
[INLONG-10736][SDK] Python sdk build script support for skipping c++ sdk build step

### LINKED ISSUE #10736: [Improve][SDK] Python dataproxy sdk build script support for skipping c++ dataproxy sdk build step
### Description

Currently the python dataproxy sdk build script must execute the build of the c++ dataproxy sdk, but the build process for the c++ sdk is slow and the user may have already built the c++ sdk, so it need to allow the user to skip this step in the build script.

### InLong Component

InLong SDK

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yfsn666**
Fixes #10736

### Motivation

Enable python sdk build script to automatically skip the c++ sdk build process when it has been compiled.

### Modifications
Modify the python sdk build script.

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh [hunk 1]
```diff
@@ -45,14 +51,19 @@ cmake --build . --config Release --target check
...
-mv ./inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-cpp ./
-rm -r ./inlong
-cd ./dataproxy-sdk-cpp
-chmod +x ./build.sh
-./build.sh
-cd $BASE_DIR
+# Build dataproxy-sdk-cpp(Optional)
+read -p "Have you already built dataproxy-sdk-cpp before? (y/n): " user_choice
+
+if [ "$user_choice" != "y" ] || [ "$user_choice" != "Y" ]; then
```
Comment (doleyzi) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh:57]: It is recommended to check whether the static library dataproxy-sdk-cpp/release/lib/libdataproxy_sdk.a has been compiled to determine whether dataproxy-sdk-cpp has been compiled.

Comment (yfsn666) [reply inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh]: fixed

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh [hunk 2]
```diff
@@ -20,6 +20,12 @@
 
 BASE_DIR=$(pwd)
 
+# Check if dataproxy-sdk-cpp directory exists in the parent directory
+if [ ! -d "../dataproxy-sdk-cpp" ]; then
```
Comment (aloyszhang) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh:24]: Plz do not use the relative path

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh [hunk 3]
```diff
@@ -45,14 +51,17 @@ cmake --build . --config Release --target check
...
-# Clone and build dataproxy-sdk-cpp
-git clone https://github.com/apache/inlong.git
-mv ./inlong/inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-cpp ./
-rm -r ./inlong
-cd ./dataproxy-sdk-cpp
-chmod +x ./build.sh
-./build.sh
-cd $BASE_DIR
+# Build dataproxy-sdk-cpp(If the dataproxy-sdk-cpp has been compiled, this step will be skipped)
+if [ ! -e "../dataproxy-sdk-cpp/release/lib/libdataproxy_sdk.a" ]; then
```
Comment (aloyszhang) [inline inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh:55]: ditto

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh (hunk 2 above)
Comment (yfsn666) [reply inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh]: fixed

File: inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh (hunk 3 above)
Comment (yfsn666) [reply inlong-sdk/dataproxy-sdk-twins/dataproxy-sdk-python/build.sh]: fixed

**Review by luchunliang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**